### PR TITLE
Flowing the nativeCodeVersion to DebuggerJitInfo

### PR DESCRIPTION
**Pull request by cshung**

### CODE REVIEW

**Comment by cshung** [comment]
@dotnet-bot Test Ubuntu16.04 arm64 Cross Checked Innerloop Build and Test

**Comment by cshung** [comment]
@dotnet-bot Test Ubuntu16.04 arm64 Cross Checked no_tiered_compilation_innerloop Build and Test

**Comment by cshung** [comment]
@dotnet-bot Test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot Test Windows_NT x64 min_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot Test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness
@dotnet-bot Test Windows_NT x86 min_opt ryujit CoreCLR Perf Tests Correctness

File: src/debug/ee/functioninfo.cpp [hunk 1]
```diff
@@ -1616,15 +1610,21 @@ DebuggerJitInfo *DebuggerMethodInfo::FindOrCreateInitAndAddJitInfo(MethodDesc* f
 
     // The DJI may already be populated in the cache, if so CreateInitAndAddJitInfo is a no-op and that is fine.
     // CreateInitAndAddJitInfo takes a lock and checks the list again, which makes this thread-safe.
+
+    CodeVersionManager::TableLockHolder lockHolder(fd->GetCodeVersionManager());
+    CodeVersionManager *pCodeVersionManager = fd->GetCodeVersionManager();
+    NativeCodeVersion nativeCodeVersion = pCodeVersionManager->GetNativeCodeVersion(fd, startAddr);
+    _ASSERTE(!nativeCodeVersion.IsNull());
```
Comment (noahfalk) [inline src/debug/ee/functioninfo.cpp]: Unforetunately EnC doesn't currently play nice with code versioning. If you edit EnC methods no new NativeCodeVersion gets generated and this search will return NULL. In that case you should be fine using the default NativeCodeVersion:

```
//some day we'll get EnC to use code versioning properly, but until then we'll get the right behavior treating all EnC versions as the default native code version.
if(nativeCodeVersion.IsNull())
{
    nativeCodeVersion = NativeCodeVersion(fd)
}
```

**Pull request merged**